hoomd/hpmc/shape_move.py

```diff
@@ -0,0 +1,322 @@
+    how to alter shape definitions
+
+    Note:
+        This class should not be instantiated by users. The class can be used
+        for `isinstance` or `issubclass` checks.
+    """
+
+    def _attach(self):
+        self._apply_param_dict()
+        self._apply_typeparam_dict(self._cpp_obj, self._simulation)
+
+
+class Callback(_HOOMDBaseObject):
+    """Base class for callbacks used in Python shape moves.
+
+    Note:
+        This class should not be instantiated by users. User-defined callbacks
+        should inherit from this class, defining a __call__ method that takes
+        a list of floats as an input and returns a shape definition
+
+    Examples::
+
+        class ExampleCallback(hoomd.hpmc.shape_move.Callback):
+            def __call__(self, params):
+                # do something with params and define verts
+                return hoomd.hpmc._hpmc.PolyhedronVertices(verts)
+    """
+
+    def __init__(self):
+        pass
+
+
+class Constant(ShapeMove):
+    """Apply a transition to a specified shape, changing a particle shape by
+    the same way every time the updater is called.
+
+    Note:
+        This is useful for calculating a specific transition probability and
+        derived thermodynamic quantities.
+
+    Args:
+        shape_params (dict): Arguments defining the shape to transition to
+
+    Examples::
+
+        mc = hoomd.hpmc.integrate.ConvexPolyhedron(23456)
+        tetrahedron_verts = [(1, 1, 1), (-1, -1, 1),
+                             (1, -1, -1), (-1, 1, -1)]
+        mc.shape["A"] = dict(vertices=tetrahedron_verts)
+        cube_verts = [(1, 1, 1), (1, 1, -1), (1, -1, 1), (-1, 1, 1),
+                      (1, -1, -1), (-1, 1, -1), (-1, -1, 1), (-1, -1, -1)])
+        constant_move = hoomd.hpmc.shape_move.Constant(shape_params=cube_verts)
+
+    Attributes:
+
+        shape_params (dict): Arguments defining the shape to transition to
+
+    See Also:
+        hoomd.hpmc.integrate for required shape parameters.
+    """
+
+    def __init__(self, shape_params):
+        self._param_dict.update(ParameterDict(shape_params=dict(shape_params)))
+
+    def _attach(self):
+        integrator = self._simulation.operations.integrator
+        if not isinstance(integrator, integrate.HPMCIntegrator):
+            raise RuntimeError("The integrator must be a HPMC integrator.")
+        if not integrator._attached:
+            raise RuntimeError("Integrator is not attached yet.")
+
+        move_cls = None
+        shapes = [
+            'Sphere', 'ConvexPolygon', 'SimplePolygon', 'ConvexPolyhedron',
+            'ConvexSpheropolyhedron', 'Ellipsoid', 'ConvexSpheropolygon',
+            'Polyhedron', 'Sphinx', 'SphereUnion'
+        ]
+        for shape in shapes:
+            if isinstance(integrator, getattr(integrate, shape)):
+                move_cls = getattr(_hpmc, 'ConstantShapeMove' + shape)
+        if move_cls is None:
+            raise RuntimeError("Integrator not supported")
```

Comment:
This bit looks the same for all classes. This should be `ShapeMove`, also you can use a list of integrator classes directly and do something like `integrator.__class__ in shapes` or use my other suggestion.